<launch>
  <arg name="camera_set"/>
  <arg name="namespace" default="$(arg camera_set)"/>
  <arg name="resize_width" default="0"/>
  <arg name="sharpen" default="0"/>
  <arg name="backend" default="opencv_nvjpeg"/>

  <node pkg="rosservice" type="rosservice" name="set_move_base_log_level" args="call --wait /move_base/set_logger_level 'ros.move_base' 'debug'" />

  <node  pkg="spinnaker_camera_driver_ros" type="camera_array.py" name="camera_array" ns="$(arg namespace)" output="screen" required="true">
    <param name="config_file" value="$(find spinnaker_camera_driver_ros)/camera_settings/$(arg camera_set).yaml"/>
    <param name="calibration_file" value="$(find spinnaker_camera_driver_ros)/calibration/$(arg camera_set).json"/>
    <param name="cache_path" value="$(find spinnaker_camera_driver_ros)/cache"/>
    <param name="preview_width" value="200"/>
    <param name="resize_width" value="$(arg resize_width)"/>
    <param name="sharpen" value="$(arg sharpen)"/>
    <param name="backend" value="$(arg backend)"/>

  </node>

</launch>

